frontend/src/pages/DocumentItemsPage.jsx

```diff
@@ -0,0 +1,324 @@
+import { useMemo, useState, useEffect } from "react";
+import { useLocation, useNavigate, useParams } from "react-router-dom";
+import DynamicTable from "@/components/table/DynamicTable";
+import PageTitle from "@/components/shared-ui/PageTitle";
+import IsError from "@/components/shared-ui/IsError";
+import { Spinner } from "@/components/ui/spinner";
+import { Button } from "@/components/ui/button";
+import { useDocumentItems } from "@/queries/documentItems";
+
+// helperi iz starog koda
+const n2 = (v) =>
+    v == null || v === "" ? null : Number(String(v).replace(",", "."));
+const r2 = (v) =>
+    v == null || Number.isNaN(v)
+        ? null
+        : Math.round((v + Number.EPSILON) * 100) / 100;
+
+const DocumentItemsPage = () => {
+    const { id } = useParams();
+    const location = useLocation();
+    const navigate = useNavigate();
+
+    // pagination state
+    const [page, setPage] = useState(1);
+    const perPage = 10;
+
+    // reset page kad promijeniš dokument ili path
+    useEffect(() => {
+        setPage(1);
+    }, [id, location.pathname]);
+
+    const rawSegment = location.pathname.split("/")[1];
```

Comment:
32-52 lines contain a complex and redundant documentType logic.
 Use useParams instead of pathname parse 
const { documentType } = useParams();
or extract into a hook: hooks/use-document-type.js